tasks/main.yml

```diff
@@ -1,2 +1,184 @@
 ---
 
+- name: Install postfix
+  package:
+    name:
+      - postfix
+      - postfix-ldap
+      - postfix-pgsql
+    state: present
+
+- name: Setup LDAP authentication
+  block:
+    - name: Install saslauthd
+      package:
+        name:
+          - libsasl2-modules
+          - sasl2-bin
+        state: present
+
+    - name: Place LDAP connection config file
+      template:
+        src: saslauthd.conf.j2
+        dest: /etc/saslauthd.conf
+
+    - name: Place SMTP SASL config file
+      copy:
+        src: sasl-smtpd.conf
+        dest: /etc/postfix/sasl/smtpd.conf
+
+    - name: Place saslauthd-postfix config file
+      template:
+        src: saslauthd-postfix.j2
+        dest: /etc/default/saslauthd-postfix
+
+    - name: Create socket directory for saslauthd
+      shell: dpkg-statoverride --add root sasl 755 /var/spool/postfix/var/run/saslauthd
+      register: dpkg_sasl
+      changed_when: "'already exists' not in dpkg_sasl.stderr"
+      failed_when: "dpkg_sasl.rc != 0 and 'already exists; aborting' not in dpkg_sasl.stderr"
+
+    - name: Add group sasl to user postfix
+      user:
+        name: postfix
+        append: True
+        groups: sasl
+
+  when: postfix_sasl_enabled
+  notify: Restart saslauthd service
+
+- name: Stop and disable saslauthd service
+  service:
+    name: saslauthd
+    state: stopped
+    enabled: no
+  when: not postfix_sasl_enabled
+  register: postfix_service_stop
+  failed_when: postfix_service_stop is failed and 'Could not find the requested service ' not in postfix_service_stop.msg
+
+- name: Setup OpenDKIM
+  block:
+    - name: Install OpenDKIM
+      package:
+        name:
+          - opendkim
+          - opendkim-tools
+        state: present
+
+    - name: Create opendkim config directory
+      file:
+        state: directory
+        dest: /etc/opendkim
+
+    - name: Create opendkim key directory
+      file:
+        state: directory
+        dest: /etc/opendkim/keys
+
+    - name: Create socket directory for opendkim
+      file:
+        state: directory
+        dest: /var/spool/postfix/var/run/opendkim
+        group: opendkim
+        mode: 0775
+
+    - name: Place opendkim config
+      copy:
+        content: "{{ postfix_opendkim_config }}"
+        dest: /etc/opendkim.conf
+
+    - name: Add group opendkim to user postfix
+      user:
+        name: postfix
+        append: True
+        groups: opendkim
+
+    - name: Place opendkim keys
```

Comment:
How does opendkim find these keys?
Is it a magic folder?
Based on what I can find online the config would need a `KeyFile` or `KeyTable` config.

Reply:
Yes, one needs to reference them in the config. That's the same as with additional files placed via `postfix_configs`.